jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextRequest.java

```diff
@@ -116,6 +118,40 @@ protected ServletContextRequest(
         addIdleTimeoutListener(_servletChannel.getServletRequestState()::onIdleTimeout);
     }
 
+    @Override
+    public Request serveAs(Request request, String path)
```

Comment:
javadoc here or on the interface would be good to describe what this is for

Reply:
I added some javadoc for this method on the interface.

Reply:
OK now I look at this more, I like it less.   It is strange to have a request implement and interface that takes another request.